pyxem/utils/calibration_utils.py

```diff
@@ -189,3 +191,146 @@ def generate_ring_pattern(
+                       ((a+c) - np.sqrt((a - c) ** 2 + b ** 2)))
+        num2 = np.sqrt(2 * (a * e ** 2 + c * d ** 2 - b * d * e + (b ** 2 - 4 * a * c) * f) *
+                       ((a+c) + np.sqrt((a - c) ** 2 + b ** 2)))
+        axis1 = abs(num1/denom)
+        axis2 = abs(num2/denom)
+
+        return np.sort([axis1, axis2])[::-1]
+
+    def ellipse_angle_of_rotation(ellipse_parameters):
+        a, b, c, d, e, f = ellipse_parameters
+        b, d, e = b/2, d/2, e/3
+        if b == 0:
+            if a > c:
+                return 0
+            else:
+                return np.pi / 2
+        else:
+            if a < c:
+                ang = .5 * invcot((a-c)/(2*b))
+                if (a < 0) == (b < 0):  # same sign
+                    ang = ang+np.pi/2
+                return ang
+            else:
+                ang = np.pi/2 + .5 * invcot((a-c)/(2*b))
+                if (a < 0) != (b < 0):
+                    ang=ang-np.pi/2
+                return ang
+
+    coords = [[], []]
+    img[mask] = 0
+    if interactive:
+        figure1 = plt.figure()
+        ax = figure1.add_subplot(111)
+        plt.imshow(i)
+        plt.text(x=-50, y=-20, s="Click to add points. Click again to remove points. You must have 5 points")
+
+        def add_point(event):
+            ix, iy = event.xdata, event.ydata
+            x_diff, y_diff = np.abs(np.subtract(coords[0], ix)), np.abs(np.subtract(coords[1], iy))
+            truth = list(np.multiply(x_diff < 10, y_diff < 10))
+            if not len(x_diff):
+                coords[0].append(ix)
+                coords[1].append(iy)
+            elif not any(truth):
+                coords[0].append(ix)
+                coords[1].append(iy)
+            else:
+                truth = list(np.multiply(x_diff < 10, y_diff < 10)).index(True)
+                coords[0].pop(truth)
+                coords[1].pop(truth)
+            print(coords)
+            ax.clear()
+            ax.imshow(img)
+            ax.scatter(coords[0], coords[1], s=10, marker="o", color="crimson")
+            figure1.canvas.draw()
+        cid = figure1.canvas.mpl_connect('button_press_event', add_point)
+        plt.show()
+
+    else:
+        coords = get_max_positions(img, num_points=num_points, radius=suspected_radius)
+    a = fit_ellipse(np.array(coords[0]), np.array(coords[1]))
+    center = ellipse_center(a)  # (x,y)
+    lengths = ellipse_axis_length(a)
+    angle = ellipse_angle_of_rotation(a)
+    print("The center is:", center)
+    print("The major and minor axis lengths are:", lengths)
+    print("The angle of rotation is:", angle)
+    cos_t = np.cos(angle)
+    sin_t = np.sin(angle)
+    R1 = [[cos_t, sin_t, 0], [-sin_t, cos_t, 0], [0, 0, 1]]
+    R2 = [[cos_t, -sin_t, 0], [sin_t, cos_t, 0], [0, 0, 1]]
+    D = [[1, 0, 0], [0, lengths[0]/lengths[1], 0], [0, 0, 1]]
+    M = np.matmul(np.matmul(R1, D), R2)
+    return center, M
+
+
+def invcot(val):
+    return (np.pi/2) - np.arctan(val)
```

Comment:
How sure are you that you want `np.arctan` rather than `np.arctan2`?

Reply:
I'm not sure.  This is stuff I pulled from my own code so i might have to double check